### DISCUSSION TITLE
Setting default signing option for commiting

### DISCUSSION THREAD

**Discussion by andr9528** [gitextensions/gitextensions#11227] [Q&A] [answered]
Hi

I might just be blind, or otherwise unlucky to not find the answer to this but...

Can the default option from the Signing dropdown, in the options during a commit, be changed?
At the moment when about to commit, it defaults to "Do not sign commit".

Have a few repositories where commits are required to be signed, for them to get past Branch Protection on the 3 main branches (master/staging/develop).
I would very much like to be able to set the option, such that it defaults to "Sign with default GPG", for select git repositories.
Is that possible?

Have already run `git config commit.gpgsign true` for one of the repositories in question, with no luck.
My environment information is as follows.

- Git Extensions 4.1.0.16698
- Build 1fe52a137acef1fbd2b653dea263b44ad5844177
- Git 2.40.1.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.20
- DPI 96dpi (no scaling)
- Portable: False
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by andr9528** [comment, marked as answer]
Sounds/Looks like #7629 will be introducing what i was asking about.
Am i right in it not being possible until that issue/feature has been solved?

**Reply by vbjay** [reply]
Correct. Even do not sign at the moment doesn't provide correct functionality if defaulted to sign which I have corrected in that feature. It is an involved feature. Needs a bit more but work has taken my time along with my frustration of @gerhardol  not getting get effective git settimg changes and that the program fails to read system level config. So I want to get back to it but later when core team gets where I am going.
